pyseir/backtest/backtest.py

```diff
@@ -0,0 +1,432 @@
+        if mf.hospital_times.size == 0:
+            mf.hospitalization_data_type = None
+
+    mf.cases_stdev, mf.hosp_stdev, mf.deaths_stdev = mf.calculate_observation_errors()
+
+    for n in range(n_retries):
+        try:
+            mf.fit()
+            if mf.mle_model:
+                break
+        except Exception as e:
+            print(e)
+
+    prediction = {}
+    prediction['new_cases'] = mf.fit_results['test_fraction'] * \
+                              np.interp(observations['times'],
+                                        mf.t_list + mf.fit_results['t0'],
+                                        mf.mle_model.results['total_new_infections'])
+    prediction['new_deaths'] = np.interp(observations['times'],
+                                         mf.t_list + mf.fit_results['t0'],
+                                         mf.mle_model.results['total_deaths_per_day'])
+
+    if mf.hospitalization_data_type is not None:
+        if mf.hospitalization_data_type is HospitalizationDataType.CUMULATIVE_HOSPITALIZATIONS:
+            predicted_hosp = (mf.mle_model.results['HGen_cumulative'] + mf.mle_model.results['HICU_cumulative'])
+            predicted_hosp = np.diff(predicted_hosp)
+        elif mf.hospitalization_data_type is HospitalizationDataType.CURRENT_HOSPITALIZATIONS:
+            predicted_hosp = mf.mle_model.results['HGen'] + mf.mle_model.results['HICU']
+        prediction['current_hosp'] = np.interp(observations['times'],
+                                               mf.t_list + mf.fit_results['t0'],
+                                               predicted_hosp * mf.fit_results['hosp_fraction'])
+    else:
+        prediction['current_hosp'] = None
+
+    prediction = pd.DataFrame(prediction, index=observations.index).dropna(axis=1, how='all')
+    
+    return prediction
+
+
+def run_backtest(fips,
+                 observations,
+                 rolling_window_size=1,
+                 prediction_window_size=7,
+                 max_observation_days_blinded=40,
+                 error_types=['nrmse', 'rmse', 'relative_error',
+                              'percentage_abs_error', 'symmetric_abs_error'],
+                 ref_date=datetime(year=2020, month=1, day=1),
+                 n_retries=3):
+    """
+    Run backtest for a given fips. Backtest blinds last n days of observations (
+    new cases, new deaths or hospitalizations) and calculates error of
+    prediction in prediction_window_size days since the last un-blinded
+    observation. Errors can be calculated for ith day of prediction where 1
+    <= i <= prediction_window_size or for the entire prediction time window,
+    depending on type of errors.
+
+    Parameters
+    ----------
+    fips: str
+        FIPS code of the state or county to run backtest.
+    rolling_window_size: int
+        Width of time window to aggregate (average) the observation and
+        prediction to calculate prediction error.
+    prediction_window_size: int
```

Comment:
Did you get the chance to use the Gaussian kernel?

Reply:
Yes, just added, thanks!